src/crystal/system/unix/socket.cr

```diff
@@ -239,28 +239,35 @@ module Crystal::System::Socket
     end
   end
 
+  @local_address_cache : ::Socket::IPAddress?
+  @remote_address_cache : ::Socket::IPAddress?
```

Comment:
I'd drop the `_cache` suffix.

```suggestion
  @local_address : ::Socket::IPAddress?
  @remote_address : ::Socket::IPAddress?
```

Reply:
I changed it to `getter` now. It should be simpler.